src/App/Pages/Vault/GroupingsPage/GroupingsPageViewModel.cs

```diff
@@ -181,9 +181,10 @@ public async Task LoadAsync()
                 return;
             }
 
+            await InitVaultFilterAsync();
             if (MainPage)
             {
-                await InitVaultFilterAsync();
+                await ShowVaultFilterIfNeededAsync();
                 PageTitle = ShowVaultFilter ? AppResources.Vaults : AppResources.MyVault;
             }
```

Comment:
Instead of splitting this into different methods I'd just change the method to 
```c#
public async Task InitVaultFilterAsync(bool shouldUpdateShowVaultFilter)
{
    ...
    if (shouldUpdateShowVaultFilter)
    {
         await Task.Delay(100);
         ShowVaultFilter = await policyService.ShouldShowVaultFilterAsync();
    }
    ...
}
```

Then here the call would just be: `await InitVaultFilterAsync(MainPage);` and on `CiphersPageViewModel` would be `await InitVaultFilterAsync(true);`